synthesizer/process/src/lib.rs

```diff
@@ -206,26 +292,64 @@ impl<N: Network> Process<N> {
     /// Returns `true` if the process contains the program with the given ID.
     #[inline]
     pub fn contains_program(&self, program_id: &ProgramID<N>) -> bool {
-        self.stacks.contains_key(program_id)
+        if self.credits.as_ref().map_or(false, |stack| stack.program_id() == program_id) {
+            return true;
+        }
+        let lock = self.stacks.lock();
+        match lock {
+            Ok(lock) => lock.contains(program_id),
+            Err(_) => false,
+        }
+    }
+
+    fn load_stack(&self, program_id: impl TryInto<ProgramID<N>>) -> Result<()> {
+        let program_id = program_id.try_into().map_err(|_| anyhow!("Invalid program ID"))?;
+        info!("Lazy loading stack for {program_id}");
```

Comment:
just double-checking that this still needs to be `INFO`-level

Reply:
0fe223518